### PR TITLE
Try to follow documentation guidelines, eliminate personal pronouns 

### PR DESCRIPTION
**Pull request by jobergum**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: en/performance/practical-search-performance-guide.md [hunk 1]
```diff
@@ -749,25 +755,31 @@ treated differently depending on having `index` or `attribute`:
...
+The reason for this default setting is that Vespa `attribute` fields can be used
+for many different aspects: [ranking](../ranking.html), [result grouping](../grouping.html),
  [result sorting](../reference/sorting.html), and finally searching/matching. 
 
-Now let us focus on the `tags` field which we defined with `attribute`,
-matching in this field will be performed using `match:word`. 
+The following section focuses on the `tags` field which we defined with `attribute`,
+matching in this field will be performed using `match:word` which is the
+default match mode for string fields with `indexing: attribute`. The
+`tags` field is as type [weightedset](../reference/schema-reference.html#type:weightedset).
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:772]:
```suggestion
default match mode for string fields with `indexing: attribute`.
The `tags` field is of type [weightedset](../reference/schema-reference.html#type:weightedset).
```

File: en/performance/practical-search-performance-guide.md [hunk 2]
```diff
@@ -749,25 +755,31 @@ treated differently depending on having `index` or `attribute`:
...
 
 <pre>
  field tags type weightedset&lt;string&gt; {
       indexing: summary | attribute
  }
 </pre>
 
-In this case, there is no index structure, searching the `attribute` 
-field is performed as a linear scan, let us do a search for a popular tag *rock*:
+Weightedset is a field type that allow representing a tag with a integer weight, which can be used for ranking.
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:780]:
```suggestion
Weightedset is a field type that allows representing a tag with an integer weight, which can be used for ranking. 
```

File: en/performance/practical-search-performance-guide.md [hunk 3]
```diff
@@ -749,25 +755,31 @@ treated differently depending on having `index` or `attribute`:
...
  field tags type weightedset&lt;string&gt; {
       indexing: summary | attribute
  }
 </pre>
 
-In this case, there is no index structure, searching the `attribute` 
-field is performed as a linear scan, let us do a search for a popular tag *rock*:
+Weightedset is a field type that allow representing a tag with a integer weight, which can be used for ranking. 
+In this case, there is no inverted index structure and matching against the `tags` field 
+is performed as a linear scan, the following scans for documents where `tags` match *rock*:
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:782]:
```suggestion
In this case, there is no inverted index structure,
and matching against the `tags` field is performed as a linear scan.
The following scans for documents where `tags` match *rock*:
```

File: en/performance/practical-search-performance-guide.md [hunk 4]
```diff
@@ -802,9 +815,9 @@ $ vespa query \
 </pre>
 </div>
 
-In all these examples searching the `tags` field, the matching is performed by a linear scan through all our documents. 
-
-We can combine the tag search with query terms searching fields that do have index structures:
+In all these examples searching the `tags` field, the matching is performed by a linear scan through all
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:818]:
```suggestion
In all these examples searching the `tags` field, the matching is done by a linear scan through all
```

File: en/performance/practical-search-performance-guide.md [hunk 5]
```diff
@@ -821,9 +834,8 @@ documents that needs to be scanned for the _tags_ constraint. This query is auto
 by the Vespa query planner. 
 
 ### Searching attribute fields using fast-search 
-As we have seen in the previous section, fields with `attribute` does per default does not
-build any [inverted index](https://en.wikipedia.org/wiki/Inverted_index) data structures. We can modify
-the tags field and add `attribute:fast-search` which will add inverted index structures for the attribute as well:
+This section adds `fast-search` to the `tags` field to speed up searches where there are no 
+other query filters which restricts the search. The schema with `fast-serch`:
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:838]:
```suggestion
other query filters which restricts the search. The schema with `fast-search`:
```

File: en/performance/practical-search-performance-guide.md [hunk 6]
```diff
@@ -876,7 +888,7 @@ $ vespa deploy --wait 300 app
 </pre>
 </div>
 
-The above will print:
+The above will print a `WARNING:
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:891]: not sure if you intended markup here or not:
```suggestion
The above will print a WARNING:
```

File: en/performance/practical-search-performance-guide.md [hunk 7]
```diff
@@ -891,8 +903,11 @@ In cluster 'tracks' of type 'search':
 Waiting up to 300 seconds for query service to become available ...
 </pre>   
 
-To enable `fast-search` Vespa content node(s) needs to be restarted to re-build the index. 
-This example uses [vespa-sentinel-cmd command tool](../reference/vespa-cmdline-tools.html#vespa-sentinel-cmd):
+To enable `fast-search` Vespa content node(s) needs to be restarted to re-build the fast-search data structures
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:906]:
```suggestion
To enable `fast-search`, content node(s) needs to be restarted to re-build the fast-search data structures
```

File: en/performance/practical-search-performance-guide.md [hunk 8]
```diff
@@ -901,15 +916,15 @@ $ docker exec vespa bash -c "/opt/vespa/bin/vespa-sentinel-cmd restart searchnod
 </pre>
 </div>
 
-Wait for the searchnode to start up using the [health state api](../reference/metrics.html):
+This step requires to wait for the searchnode, use [health state api](../reference/metrics.html):
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:919]: consider content node?
```suggestion
This step requires waiting for the searchnode, use the [health state api](../reference/metrics.html):
```

File: en/performance/practical-search-performance-guide.md [hunk 9]
```diff
@@ -923,7 +938,7 @@ We want the status code to flip to up before querying again:
 $ sleep 60
 </pre>
 
-Once up, we can try our multi-tag search again - now using the `fast-search` attribute:
+Once up, execute the `tags`query again:
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:941]:
```suggestion
Once up, execute the `tags` query again:
```

File: en/performance/practical-search-performance-guide.md [hunk 10]
```diff
@@ -939,23 +954,29 @@ Now the `querytime` will be a few milliseconds since Vespa has built index struc
...
 [searching multi-value fields](../searching-multi-valued-fields.html).
 
 ## Multi-valued query operators
 
-In this section we look at [multi-value query operators](../multivalue-query-operators.html) 
-and their query performance characteristics. 
-Many real-world use cases needs search using structured queries with many values. 
+This section covers [multi-value query operators](../multivalue-query-operators.html) 
+and their query performance characteristics. Many real-world search and recommendation use cases 
+involves structured multi-valued queries.
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:965]:
```suggestion
involve structured multi-valued queries. 
```

File: en/performance/practical-search-performance-guide.md [hunk 11]
```diff
@@ -1006,10 +1027,9 @@ schema track {
 }
 </pre>
 
-The `dotProduct`and `wand` query operators produces a rank feature called
- [rawScore(name)](../reference/rank-features.html#rawScore(field)), which calculates
- the dot product between the query and document weights. We use 
- `rank-feature` as our first-phase rank expression. 
+The `dotProduct`and `wand` query operators produces a `rank feature` called
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1030]:
```suggestion
The `dotProduct`and `wand` query operators produce a `rank feature` called
```

File: en/performance/practical-search-performance-guide.md [hunk 12]
```diff
@@ -1112,30 +1147,39 @@ For larger document collections, the *wand* query operator can significantly
...
+field type. The `weightedset` field type only supports integer weights. The next section
+covers tensors that support more floating point number types. 
 
 ## Tensor Computations
-In the previous sections we looked at matching and where query matching query operator also produced rank features which we could
-use to influence the order of the hits returned. In this section we look at [tensor computations](../tensor-examples.html) 
-using [tensor expressions](../tensor-user-guide.html). Tensor computations can be used to calculate dense dot products, sparse
-dot products, matrix multiplication, neural networks and more. 
+The previous sections covered matching and where query matching query operators 
+also produced rank features which could use to influence the order of the hits returned.
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1160]:
```suggestion
also produced rank features which could be used to influence the order of the hits returned. 
```

File: en/performance/practical-search-performance-guide.md [hunk 13]
```diff
@@ -1112,30 +1147,39 @@ For larger document collections, the *wand* query operator can significantly
...
-For each of the track documents we have a field named `similar` which is of type `tensor`
-with one named *mapped* dimension. *Mapped* tensors can be used to represent sparse feature representations:
+Tensor computations can be used to calculate dense dot products, sparse
+dot products, matrix multiplication, neural networks and more. Tensor computations can be performed 
+on documents that are retrieved by the query matching operators. The only exception to this is
+dense single order tensors (vectors) where Vespa also supports "matching" using [(approximate) nearest
+neighbor search](..//approximate-nn-hnsw.html). 
+
+
+The `track` schema was defined a `similar` tensor field with one named *mapped* dimension.
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1171]:
```suggestion
The `track` schema was defined with a `similar` tensor field with one named *mapped* dimension. 
```

File: en/performance/practical-search-performance-guide.md [hunk 14]
```diff
@@ -1112,30 +1147,39 @@ For larger document collections, the *wand* query operator can significantly
...
+Tensor computations can be used to calculate dense dot products, sparse
+dot products, matrix multiplication, neural networks and more. Tensor computations can be performed 
+on documents that are retrieved by the query matching operators. The only exception to this is
+dense single order tensors (vectors) where Vespa also supports "matching" using [(approximate) nearest
+neighbor search](..//approximate-nn-hnsw.html). 
+
+
+The `track` schema was defined a `similar` tensor field with one named *mapped* dimension. 
+*Mapped* tensors can be used to represent sparse feature representations, similar
+to our `weightedset` field, but in a more generic way, and here using `float` to represent
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1173]:
```suggestion
to the `weightedset` field, but in a more generic way, and here using `float` to represent
```

File: en/performance/practical-search-performance-guide.md [hunk 15]
```diff
@@ -1112,30 +1147,39 @@ For larger document collections, the *wand* query operator can significantly
...
+the tensor cell value. 
 
 <pre>
 field similar type tensor&lt;float&gt;(trackid{}) {
       indexing: summary | attribute
 }
 </pre>
 
-Let us look at one of the tracks in our collection:
+Inspecting one document, using the vespa-cli (Wraps [Vespa document/v1 api](../document-v1-api-guide.html)).
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1182]:
```suggestion
Inspecting one document, using the vespa-cli (Wraps [Vespa document/v1 api](../document-v1-api-guide.html)):
```

File: en/performance/practical-search-performance-guide.md [hunk 16]
```diff
@@ -1144,7 +1188,7 @@ $ vespa document get id:music:track::TRQIQMT128E0791D9C
 </pre>
 </div>
 
-We leave out some of the tags and the tensor cells from the output, but below is a summary:
+Returns the following
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1191]:
```suggestion
Returns:
```

File: en/performance/practical-search-performance-guide.md [hunk 17]
```diff
@@ -1247,42 +1292,42 @@ Which returns:
 }
 {% endhighlight %}</pre>
 
-Now, if a user has listened to these three tracks in a session:
+Given a single track, one could just retrieve the document and display the offline computed similar tracks, but,
+if a user has listened to multiple tracks in a real time session, one could use a sparse dot product
+between the user recent activity and the track similarity fields. For example, listening
+to the following tracks
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1298]:
```suggestion
to the following tracks:
```

File: en/performance/practical-search-performance-guide.md [hunk 18]
```diff
@@ -1247,42 +1292,42 @@ Which returns:
...
-computation: 
+Could be represented as a query tensor `query(user_liked)` and passed with the query request like this:
 
 <pre>
-rank-profile similar {
-    first-phase {
-        expression: sum(attribute(similar) * query(user_liked))
-    }
-}
+ranking.features.query(user_liked)={{trackid:TRUAXHV128F42694E8}:1.0,{trackid:TRQIQMT128E0791D9C}:1.0,{trackid:TRGVORX128F4291DF1}:1.0}
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1307]: might have to add raw / endraw on the pre

File: en/performance/practical-search-performance-guide.md [hunk 19]
```diff
@@ -1247,42 +1292,42 @@ Which returns:
...
 </pre>
 
-The `similar` rank profile with defines a sparse tensor dot product between the `query(user_liked)` query tensor and the 
-the `attribute(similar)` document tensor. See [tensor user guide](../tensor-user-guide.html) for more on tensors.
-We could also use [phased ranking](../phased-ranking.html) to reduce the computational complexity:
+The tensor type of this query tensor was introduced in previous sections
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1310]:
```suggestion
The tensor type of this query tensor was introduced in previous sections:
```

File: en/performance/practical-search-performance-guide.md [hunk 20]
```diff
@@ -1247,42 +1292,42 @@ Which returns:
...
+
+<pre data-test="file" data-path="app/search/query-profiles/types/root.xml">
+&lt;query-profile-type id=&quot;root&quot; inherits=&quot;native&quot;&gt;
+    &lt;field name=&quot;ranking.features.query(user_liked)&quot; type=&quot;tensor&amp;lt;float&amp;gt;(trackid{})&quot; /&gt;
+&lt;/query-profile-type&gt;
+</pre>
+
+Both the document tensor and the query tensor are defined with `trackid{}` as the *named* *mapped* dimension. The 
+sparse tensor dot product can then be expression in a `rank-profile` like this:
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1319]:
```suggestion
sparse tensor dot product can then be expression in a `rank-profile`:
```

File: en/performance/practical-search-performance-guide.md [hunk 21]
```diff
@@ -1479,10 +1528,9 @@ The above query produces the following result:
 }
 {% endhighlight %}</pre>
 
-Now we can notice that with the `not` filter, we retrieved 95,663 documents
-as our 3 tracks previously *liked*, where filtered out. 
+This query retrieves 95,663 document and the three tracks previously *liked*, where removed from the result.
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1531]:
```suggestion
This query retrieves 95,663 documents, and the three tracks previously *liked* were removed from the result.
```

File: en/performance/practical-search-performance-guide.md [hunk 22]
```diff
@@ -1479,10 +1528,9 @@ The above query produces the following result:
...
 
-We can also add tag filters, here we filter the recommended by `tags:popular`, by filtering we reduce the complexity of the
+The following example filters by a tags query, `tags:popular`, which reduces the complexity of the
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1533]:
```suggestion
The following example filters by a tags query, `tags:popular`, reducing the complexity of the
```

File: en/performance/practical-search-performance-guide.md [hunk 23]
```diff
@@ -1605,21 +1656,19 @@ $ vespa query \
 
 The `querytime` dropped to 40 ms instead of 120 ms without the `fast-search` option. 
 See also [performance considerations](../tensor-user-guide.html#performance-considerations)
-when using tensor expressions and especially the tensor cell precision used.
-Vespa supports both `int8`, `bfloat16`, `float` and
-`double` precision cell types.  
+when using tensor expression. Vespa supports both `int8`, `bfloat16`, `float` and
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1659]:
```suggestion
when using tensor expression. Vespa supports `int8`, `bfloat16`, `float` and
```

File: en/performance/practical-search-performance-guide.md [hunk 24]
```diff
@@ -1605,21 +1656,19 @@ $ vespa query \
...
-need to change our `services.xml`. 
+So far in this guide all search queries and ranking computations have been performed using 
+single threaded execution. To enable multi-threaded execution, a setting needs to be 
+added to `services.xml`. 
 Multi-threaded search and ranking can improve query latency significantly and make better
 use of multi-cpu core architectures. Search and ranking has evolved since 1998 and by 
-using multi-threaded execution we can significantly reduce serving latency. 
-In many deployment systems, low latency is critical.  
+using multi-threaded execution, serving latency can be reduced significantly.  
+In many deployed search and recommendation systems, low serving latency is critical.
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1669]:
```suggestion
In many search and recommendation systems, low serving latency is critical.  
```

File: en/performance/practical-search-performance-guide.md [hunk 25]
```diff
@@ -1778,38 +1828,39 @@ $ vespa deploy --wait 300 app
...
 limits query concurrency as more threads will be occupied
 per query. Read more in [Vespa sizing guide:reduce latency with 
 multi-threaded search](sizing-search.html#reduce-latency-with-multi-threaded-per-search-execution).
-Note also that the number of search threads versus number of threads per search limits
-the number of **concurrent** queries a content node can handle. 
 
 ## Advanced range search with hitLimit  
 
-Vespa has an advanced query operator that allows you to select the 
+Vespa has an advanced query operator that allows select the
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:1854]:
```suggestion
Vespa has an advanced query operator that allows selecting the 
```

File: en/performance/practical-search-performance-guide.md [hunk 26]
```diff
@@ -1998,7 +2049,8 @@ Deploy the application again :
 $ vespa deploy --wait 300 app
 </pre>
 </div>
-Adding a new field does not require any restart, but we need to populate the popularity field:
+
+Adding a new field does not require any restart, apply the partial updates by:
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:2053]:
```suggestion
Adding a new field does not require a restart, apply the partial updates by:
```

File: en/performance/practical-search-performance-guide.md [hunk 27]
```diff
@@ -2008,8 +2060,8 @@ $ ./vespa-feed-client-cli/vespa-feed-client \
 </pre>
 </div>
 
-With that feed job completed, we can select 5 (`hitLimit`) tracks with the highest popularity by 
-using the [range()](../reference/query-language-reference.html) query operator:
+With that feed job completed, it is possible to use select the five tracks with the highest popularity by
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:2063]:
```suggestion
With that feed job completed, it is possible to select the five tracks with the highest popularity by 
```

File: en/performance/practical-search-performance-guide.md [hunk 28]
```diff
@@ -2051,22 +2094,24 @@ $ vespa query \
...
+The range search with `hitLimit` can be used for cases where one wants to select efficiently *top-k* of a 
+single valued numeric `attribute` with `fast-search`. Some use cases which can be efficiently implemented
+with it:
 
-- We can use it to for example to only run computations over the 1,000 most recent documents 
-using a long to represent a timestamp
-- We can use it as above to compute and rank the most popular documents.
+- Run ranking computations over the most recent documents 
+using a long to represent a timestamp (e.g, using Unix epoch).
+- Compute personalization tensor expressions over pre-selected content, e.g using popularity.
```
Comment (kkraune) [inline en/performance/practical-search-performance-guide.md:2106]:
```suggestion
using a `long` to represent a timestamp (e.g., using Unix epoch).
- Compute personalization tensor expressions over pre-selected content, e.g. using popularity.
```

**Comment by jobergum** [comment]
Thanks once again, Kristian, for the great reviews!

**Pull request merged**